### ISSUE TITLE
Mix of public and private suffixes

### ISSUE THREAD

**Issue by sebix** [barseghyanartur/tld#27]
As the public suffix list is used to find the "TLD" (which is not the TLD in many cases), also private domains are now returned as TLD. Have a look at the list: https://publicsuffix.org/list/effective_tld_names.dat or https://publicsuffix.org/list/public_suffix_list.dat.

It is split into two sections, the public and the private suffixes. In the readme there is perfect example about how this get's wrong. The private suffixes can be submitted by every domain owner.

Eg. `co.uk` is a public suffix. A private suffix is e.g. `appspot.com`.

I see two issues here:
1. what you consider to be a "tld" is a (public or private) suffix plus the next domain name below. But that is not the "top level domain", it is first or second level (below the tld)! I think that deviation from standards should be documented *much* better.
2. you mix private and publix suffixes. So you treat e.g. `appspot.com` the same as `com` as equally valid suffixes! Have a look at the private suffix section in the list to see how much private domains are there.

**Comment by barseghyanartur** [comment]
@sebix:

Your concerns make sense. However, I don't think regular user would need that much detail, still, there should be a way of narrowing the match and expected based on certain input arguments, in such a way, that default behaviour would stay intact (this package is used a lot; backwards compatibility is a must).

**Comment by barseghyanartur** [comment]
@sebix:

I agree about blogspot.com not being the same level tld as com. There's no question about that. I think there should be some transitional situation/release, where the new behaviour is introduced first as non-default, with a clear message about deprecation and in some version later on, the new behaviour is set as a default, still leaving a space for turning on the old behaviour.

I've added it to the TODOs. It will be addressed soon. 

Thanks again. Your (critical) opinion is much appreciated.

**Comment by Yomguithereal** [comment]
On a side note, if one would like to implement that, does the Mozilla files holds necessary metadatato categorize the listed TLDs as public/private?

**Comment by barseghyanartur** [comment]
@Yomguithereal:

Regarding the implementation. This is how I think it shall be implemented.

At the moment we have only one variable ``tld_names`` to store data in. The suggestion is to split them into two separate sets:

- ``public_tld_names``
- ``private_tld_names``

The ``get_tld`` function will get two additional (optional arguments).

```python
def get_tld(url,
                  active_only=False,
                  fail_silently=False,
                  as_object=False,
                  fix_protocol=False,
                  search_public=True,
                  search_private=True):
    """Implementation"""
    # TODO
```

The ``search_public`` and ``search_private`` flags are to indicate where to search. If both are given, search in both until the first match is found.

If ``search_public`` is set to False, we don't search in public domains.
If ``search_private`` is set to False, we don't search in private domains.

Default values for both flags are True.

**Comment by barseghyanartur** [comment]
@Yomguithereal:

Any progress on this?

**Comment by Yomguithereal** [comment]
Just opened a PR.

**Comment by barseghyanartur** [comment]
@Yomguithereal:

Awesome, thanks!

**Comment by barseghyanartur** [comment]
Solved in branch [v0.8](https://github.com/barseghyanartur/tld/tree/v0.8). To be released in 0.8.

**Comment by barseghyanartur** [comment]
Released in 0.8.
